File: Week_09/java/src/_344_ReverseString.java
/**
 * 第一遍：2020/07/15周三 ✅
 * 第二遍：2020/07/16周四 ✅
 * 第三遍：2020/07/08周四
 * 第四遍：2020/07/13周一
 * 第三遍：2020/06/29周一
 * 第四遍：2020/07/05周日
 */
class _344_ReverseString {
    public void reverseString(char[] s) {
        if (s == null || s.length == 0) return;
        int left = 0, right = s.length - 1;
        while (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }
}

File: Week_09/kotlin/src/_344_ReverseString.kt
fun reverseString(s: CharArray): Unit {
    if (s.isEmpty()) return
    var l = 0; var r = s.size - 1
    while (l < r) {
        val temp = s[l]; s[l] = s[r]; s[r] = temp; l++; r--
    }
}